datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxDetachService.java

```diff
@@ -0,0 +1,95 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.auth.crn.CrnResourceDescriptor;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareCrnGenerator;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+
+/**
+ * This class detaches the data lake from the environment.
+ * <p>
+ * It is achieved by renaming name and cluster CRN. CLuster associated with the stack is also renamed as well.
+ * After this operation is performed, the stack associated with this cluster is also renamed so that DL name and stack same are the same.
+ * <p>
+ */
+@Component
+public class SdxDetachService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxDetachService.class);
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
+
+    @Inject
+    private RegionAwareCrnGenerator regionAwareCrnGenerator;
+
+    @Inject
+    private SdxDetachNameGenerator sdxDetachNameGenerator;
+
+    @Inject
+    private SdxAttachDetachUtils sdxAttachDetachUtils;
+
+    /**
+     * Detaches the internal SDX cluster by assigning it a new "detached" name and CRN.
+     */
+    public SdxCluster detachCluster(Long sdxID) {
+        LOGGER.info("Started detaching SDX cluster with ID: {}.", sdxID);
+        sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.STOPPED, "Datalake detach in progress.", sdxID);
+
+        SdxCluster cluster = sdxClusterRepository.findById(sdxID)
```

Comment:
you should use `SdxService#getById` instead

Reply:
Done.